flax/linen/module.py

```diff
@@ -534,7 +534,13 @@ def init(self, rngs, *args, method=None, **kwargs):
 
   @property
   def variables(self):
-    return self.scope.variables()
```

Comment:
It's possible the user also tries to access variables before the scope is defined. Should we check for that too?